### ISSUE TITLE
Add a POST /password/forgot endpoint

### ISSUE THREAD

**Issue by begedin** [code-corps/code-corps-api#730]
# We need to Add a `POST /password/forgot` endpoint

* Requires an email parameter
* Looks for a user with the specified email
* Generates an `AuthToken` model defined in #729
* Generates an email with a link which **takes the user to the ember app route to reset password** - this could be difficult, there's a separate issue #733
* Sends that email to the user
* Endpoint should return a 200 either way, even if the submitted email was wrong and the password reset link was not actually sent.

## Subtasks
- [x] add endpoint to router
- [x] add `PasswordController` with `forgot_password` function
- [x] ensure endpoint returns a 200 OK, either way
- [ ] add behavior described above to endpoint - this part is partially blocked by #729, but the rest can be done
- [x] add controller tests for endpoint

## References

Progress on: #498
